src/java.base/share/classes/java/lang/System.java

```diff
@@ -2594,6 +2594,14 @@ public void inflateBytesToChars(byte[] src, int srcOff, char[] dst, int dstOff,
                 StringLatin1.inflate(src, srcOff, dst, dstOff, len);
             }
 
+            public byte stringCoder(String s) {
```

Comment:
I think it'd be better to do `boolean isLatin1(String s)` here. It's still somewhat leaky, but keeps things a bit more high-level, containing more of the low-level details. Seems all uses basically just checks `coder == 0` anyhow.